### PR TITLE
[#1473] Make deps resolve paths relative to bb.edn

### PR DESCRIPTION
**Pull request by lispyclouds**
- [ ] tests
- [x] tests for deps-root
- [x] changelog

### CODE REVIEW

**Comment by lispyclouds** [comment]
Has the same flaky failures, but seems to be okay.

**Comment by borkdude** [comment]
It would be nice to have a test for this. Filling out the PR template would be nice too which reminds you of this.

File: src/babashka/impl/deps.clj [hunk 1]
```diff
@@ -89,8 +89,12 @@
                             "-Sdeps-file" "") ;; we reset deps file so the local deps.edn isn't used
                  args (if force (cons "-Sforce" args) args)
                  args (concat args [(str "-A:" (str/join ":" (cons ":org.babashka/defaults" aliases)))])
-                 cp (with-out-str (binding [deps/*env* env
-                                            deps/*extra-env* extra-env]
+                 deps-root (:deps-root @bb-edn)
```
Comment (borkdude) [inline src/babashka/impl/deps.clj:92]: What I meant was that deps-root is used higher up, so pulling out some levels higher lets us re-use it.

**Comment by lispyclouds** [comment]
@borkdude if i add a tests for the relative to bb.edn with something like
```clojure
(spit config
      '{:paths ["src"]
         :deps  {local/dep {:local/root "local-dep"}}
         :tasks {cp (prn (babashka.classpath/get-classpath))}})
(testing "relative paths in deps should be relative to bb.edn"
  (let [root (fs/create-dir (fs/file dir "local-dep"))
        _ (spit (str (fs/file root "deps.edn")) {})
        out     (bb "--config" config "cp")
        entries (cp/split-classpath out)]
    (is (= (fs/parent f) (fs/parent (first entries))))))
```
this fails to create the cache folder with (ran on Mac):
```
==== Testing JVM version

lein test babashka.bb-edn-test
=== deps-root-tests

Error building classpath. Can't create directory: /private/Users/rahul.de/.clojure/.cpcache
Tests failed.
```

Any suggestions?

**Pull request merged**